tests/test_repository.py

```diff
@@ -1113,35 +1129,48 @@ def test_auto_track_large_files(self):
         # This content is 20MB (over 10MB)
         large_file = [100] * int(4e6)
 
+        # This content is binary (contains the null character)
+        binary_file = "\x00\x00\x00\x00"
```

Comment:
I would consider moving this to another test to keep this test focused in large files (or rename this test, although I think different tests would be better)

Reply:
Good point! Let's keep as is then :smile: